tools/compiletest/src/runtest.rs

```diff
@@ -387,6 +392,99 @@ impl<'test> TestCx<'test> {
         }
     }
 
+    /// Modify the results section of the output from the new format to the old
+    /// format so that it works for existing `expected` files that are based on
+    /// the old format
+    fn reformat_output(output: &str) -> String {
+        // Process the output on a line-by-line basis
+        let lines: Vec<String> = output.split('\n').map(|s| s.to_string()).collect();
+        let mut adapted_lines: Vec<String> = Vec::new();
+
+        // 1. Copy all lines until and including the "RESULTS:" line, which
+        // marks the beginning of the results section of the output
+        let mut line_iter = lines.iter();
+        while let Some(line) = line_iter.next() {
+            adapted_lines.push(line.clone());
+            if adapted_lines.last().unwrap() == "RESULTS:" {
+                break;
+            }
+        }
+
+        if line_iter.clone().peekable().peek().is_none() {
+            // iterated over all the lines, so this is not the new format
```

Comment:
This should error out saying that the output format did not match the one you are expecting.